src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -542,4 +585,57 @@ TokenRefresher create(FirebaseApp app) {
       STOPPED
     }
   }
+
+  private static class DefaultConfigOptions{
+    private String databaseURL;
+    private String storageBucket;
+    private Map<String, Object> databaseAuthVariableOverride;
+    private String projectId;
+
+    private DefaultConfigOptions() {}
+
+    private static DefaultConfigOptions getOptionsFromEnv() {
+      String defaultConfig = System.getenv(DEFAULT_CONFIG_OPTIONS_VAR);
+      if (defaultConfig == null || defaultConfig.isEmpty()) {
+        return new DefaultConfigOptions();
+      }
+      Gson gson = new Gson();
```

Comment:
Can we not use Gson here? Right now we don't have a direct dependency on it, and we should avoid taking one if possible. Lets use `org.json` or `google-api-client` JSON support here (I prefer latter).